public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import * as H from 'history';
+
+import { locationService } from '@grafana/runtime';
+import { SceneObject, SceneObjectBase, SceneObjectState, sceneUtils, VizPanel } from '@grafana/scenes';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { getDashboardUrl } from '../utils/utils';
+
+import { PanelEditorRenderer } from './PanelEditorRenderer';
+
+export interface PanelEditorState extends SceneObjectState {
+  panel: VizPanel;
+  controls?: SceneObject[];
+  isDirty?: boolean;
+  /** Panel to inspect */
+  inspectPanelId?: string;
+  /** Scene object that handles the current drawer */
+  drawer?: SceneObject;
+  getDashboard(): DashboardScene;
+}
+
+export class PanelEditor extends SceneObjectBase<PanelEditorState> {
+  static Component = PanelEditorRenderer;
+
+  public getPageNav(location: H.Location) {
+    return {
+      text: 'Edit panel',
+      parentItem: this.state.getDashboard().getPageNav(location),
+    };
+  }
+
+  public onDiscard = () => {
+    // Open question on what to preserve when going back
+    // Preserve time range, and variables state (that might have been changed while in panel edit)
+    // Preserve current panel data? (say if you just changed the time range and have new data)
+    this.navigateBackToDashboard();
+  };
+
+  public onApply = () => {
+    // TODO handle applying changes to source dashboard state
+    this.navigateBackToDashboard();
+  };
+
+  public onSave = () => {
+    // Apply change
+    // Open dashboard save drawer
+  };
+
+  private navigateBackToDashboard() {
+    locationService.push(
+      getDashboardUrl({
+        uid: this.state.getDashboard().state.uid,
+        currentQueryParams: locationService.getLocation().search,
+      })
+    );
+  }
+}
+
+export function buildPanelEditScene(dashboard: DashboardScene, panel: VizPanel) {
+  const panelClone = panel.clone();
+  const dashboardStateCloned = sceneUtils.cloneSceneObjectState(dashboard.state);
```

Comment:
Thinking about this - do we need to clone the dashboard state? What we are technically doing in panel edit (an any object editor in the future) we are indirectly editing a dashboard. Wouldn't it be enough for us to pass the dashboard object to the edit mode, and call `onEnterEditMode` when PanelEditor activates?

Reply:
Yes we need to clone it, otherwise, we would change the parent of those scene objects (SceneVariableSet, $timeRange etc) 

> What we are technically doing in panel edit (and any object editor in the future) we are indirectly editing a dashboard. 

Sort of, but if we are to make this a separate route I think it should be more standalone. But maybe if we want to keep it more similar to how it is today (and how annotations, variables editing ) will work then we should have it remain a `editPanel?` query param and handle it in DashboardScene and the same route (DashboardScenePage).